src/main/java/com/jumani/rutaseg/controller/OrderController.java

```diff
@@ -174,4 +178,60 @@ private OrderResponse createOrderResponse(Order order) {
                 customsDataResponse
         );
     }
+
+    @GetMapping("/search")
+    public ResponseEntity<List<OrderResponse>> search(
+            @RequestParam(value = "pema", required = false) Boolean pema,
+            @RequestParam(value = "transport", required = false) Boolean transport,
+            @RequestParam(value = "port", required = false) Boolean port,
+            @RequestParam(value = "arrivalDateFrom", required = false) LocalDate arrivalDateFrom,
+            @RequestParam(value = "arrivalDateTo", required = false) LocalDate arrivalDateTo,
+            @RequestParam(value = "arrivalTimeFrom", required = false) LocalTime arrivalTimeFrom,
+            @RequestParam(value = "arrivalTimeTo", required = false) LocalTime arrivalTimeTo,
+            @RequestParam(value = "clientId", required = false) Long clientId,
+            @RequestParam(value = "status", required = false) OrderStatus status,
+            @RequestParam(value = "pageSize", required = false, defaultValue = "10") Integer pageSize,
+            @Session SessionInfo session
+    ) {
+        final Long theClientId;
+        if (!session.admin()) {
+            theClientId = session.id();
+        } else {
+            theClientId = clientId;
+        }
+        final int thePageSize;
+        if (!session.admin()) {
+            thePageSize = 1;
+        } else {
+            thePageSize = pageSize;
+        }
+
+        if (Objects.nonNull(arrivalDateFrom) && Objects.nonNull(arrivalDateTo) && arrivalDateFrom.isAfter(arrivalDateTo)) {
```

Comment:
Viendo lo que planteaste pienso que podríamos permitir que from y to sean iguales, en ese caso estaríamos diciendo "traeme las órdenes de este día", bien ahi!